packages/common/src/ObjectUtils.ts

```diff
@@ -44,4 +44,18 @@ export const createObjectFromDefaultProps = (overrideProps: ObjectType, defaultP
   return newObject;
 };
 
-module.exports = { createObjectFromDefaultProps };
+/**
+ * @notice Given an`object`, and a `mapFn`, returns a new Object having applied the mapping function to each element in
+ * the `object`. Analogous to Array.prototype.map(), but for Objects.
+ * @param {object} the Object to map over.
+ * @param {mapFn} the mapping function to apply to each element in the `object`.
+ * @returns {object} new Object with the same keys as the `object`, but each value is the result of applying the mapFn.
+ */
+export const objectMap = <T, U>(object: T, mapFn: (arg: T[keyof T]) => U): { [Property in keyof T]: U } => {
```

Comment:
@chrismaree took a little extra work, but this should be working now. I have trouble intuiting these generic functions without compiling them myself. Hope you don't mind, I just pushed up the change to fix it.

Reply:
The result of this function working is typescript being able to correctly infer return types as shown here:

<img width="402" alt="Screen Shot 2021-12-09 at 2 49 53 PM" src="https://user-images.githubusercontent.com/11791551/145465933-3f47a766-51a7-4786-a008-dad6defc1e3e.png">

<img width="625" alt="Screen Shot 2021-12-09 at 2 50 00 PM" src="https://user-images.githubusercontent.com/11791551/145465860-27f4b669-1fae-4654-b020-21b22902d3b3.png">


Note: it's a little confusing bc the input and prop have the same name, but the second image is the property that you're setting in the log object. This isn't too impactful here, but it is more meaningful if you were to take these output values and use them for something that required the compiler to know that they're strings.

Reply:
thanks @mrice32 ! I see what you did there, pretty clever using `keyof`!